### PR TITLE
Deprecate `Nullable` `zip` method in favour of DSL

### PR DESCRIPTION
**Pull request by gutiory**
This PR implements the [issue](https://github.com/arrow-kt/arrow/issues/2967)
<s>Unfortunately, the replacement is not working with implicit function parameters 😞 <s>.

```kotlin
Nullable.zip(a) { it + 1 }
```
Is replaced by 
```kotlin
nullable<Int> { it + 1 }
```
Instead of 
```kotlin
nullable { x.bind() + 1 }
```

### CODE REVIEW

File: arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Nullable.kt [hunk 1]
```diff
@@ -10,83 +11,134 @@ import kotlin.jvm.JvmStatic
 public object Nullable {
 
   @JvmStatic
+  @Deprecated(
+    "Prefer using the inline result DSL",
+    ReplaceWith(
+      "nullable<A> { fn(a.bind<A>()) }", "arrow.core.raise.nullable"
```
Comment (nomisRev) [inline arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Nullable.kt:17]:
```suggestion
      "a?.let(f)"
```

File: arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Nullable.kt [hunk 2]
```diff
@@ -10,83 +11,134 @@ import kotlin.jvm.JvmStatic
...
+  ): R? {
     contract { callsInPlace(fn, InvocationKind.AT_MOST_ONCE) }
     return zip(a, b, c, d, e, f, g, h, Unit) { a, b, c, d, e, f, g, h, _ -> fn(a, b, c, d, e, f, g, h) }
   }
 
   @JvmStatic
+  @Deprecated(
+    "Prefer using the inline result DSL",
+    ReplaceWith(
+      "nullable<A, B, C, D, E, F, H, I> { fn(a.bind<A>(), b.bind<B>(), c.bind<C>(), d.bind<D>(), e.bind<E>(), f.bind<F>(), g.bind<G>(), h.bind<H>(), i.bind<I>()) }",
```
Comment (franciscodr) [inline arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Nullable.kt:121]:
```suggestion
      "nullable<A, B, C, D, E, F, G, H, I> { fn(a.bind<A>(), b.bind<B>(), c.bind<C>(), d.bind<D>(), e.bind<E>(), f.bind<F>(), g.bind<G>(), h.bind<H>(), i.bind<I>()) }",
```

File: arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Nullable.kt [hunk 3]
```diff
@@ -10,83 +11,134 @@ import kotlin.jvm.JvmStatic
...
   ): R? {
     contract { callsInPlace(fn, InvocationKind.AT_MOST_ONCE) }
     return zip(a, b, c, d, e, f, g, h, i, Unit) { a, b, c, d, e, f, g, h, i, _ -> fn(a, b, c, d, e, f, g, h, i) }
   }
 
   @JvmStatic
+  @Deprecated(
+    "Prefer using the inline result DSL",
+    ReplaceWith(
+      "nullable<A, B, C, D, E, F, I, J> { fn(a.bind<A>(), b.bind<B>(), c.bind<C>(), d.bind<D>(), e.bind<E>(), f.bind<F>(), g.bind<G>(), h.bind<H>(), i.bind<I>(), j.bind<J>()) }",
```
Comment (franciscodr) [inline arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Nullable.kt:136]:
```suggestion
      "nullable<A, B, C, D, E, F, G, H, I, J> { fn(a.bind<A>(), b.bind<B>(), c.bind<C>(), d.bind<D>(), e.bind<E>(), f.bind<F>(), g.bind<G>(), h.bind<H>(), i.bind<I>(), j.bind<J>()) }",
```

**Review by franciscodr — COMMENTED** [review]
Two minor changes

File: arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Nullable.kt (hunk 1 above)
Comment (gutiory) [reply arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Nullable.kt]: Thanks, @nomisRev ! See [`2e1bc4f` (#2983)](https://github.com/arrow-kt/arrow/pull/2983/commits/2e1bc4fccd08be10fa0e8870c4598625965af3e6)

File: arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Nullable.kt (hunk 2 above)
Comment (gutiory) [reply arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Nullable.kt]: Thanks! See [`2e1bc4f` (#2983)](https://github.com/arrow-kt/arrow/pull/2983/commits/2e1bc4fccd08be10fa0e8870c4598625965af3e6)

File: arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Nullable.kt (hunk 3 above)
Comment (gutiory) [reply arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Nullable.kt]: Thanks! See [`2e1bc4f` (#2983)](https://github.com/arrow-kt/arrow/pull/2983/commits/2e1bc4fccd08be10fa0e8870c4598625965af3e6)

**Review by franciscodr — APPROVED** [review]
Thanks, @gutiory!

**Comment by rafaparadela** [comment]
Task linked: [CU-865byuun4 Deprecate Nullable zip method in favour of DSL](https://app.clickup.com/t/865byuun4)

**Review by nomisRev — APPROVED** [review]
Thank you @gutiory 🙌

**Pull request merged**